warehouse/schema.go

```diff
@@ -1,53 +1,90 @@
 package warehouse
 
-import "fmt"
+import (
+	"reflect"
+	"time"
+)
+
+type Field interface {
+	Name() string
+	DataType() reflect.Type
+	IsTime() bool
+}
+
+type dataField struct {
+	name     string
+	dataType reflect.Type
+	isTime   bool
+}
 
-type Field struct {
-	Name   string
-	DBType string
+func (f *dataField) Name() string {
+	return f.name
 }
 
-func (f Field) String() string {
-	return fmt.Sprintf("%s %s", f.Name, f.DBType)
+func (f *dataField) DataType() reflect.Type {
+	return f.dataType
```

Comment:
It might be more clear to have this method return `reflect.TypeOf(time.Now())` when appropriate and get rid of `IsTime()`

Reply:
I apologize, I had decided that since you have to check for a time data type on every field on every row, I did not want to always repeat that work.  I was refactoring to do the check once when the Field slices were first created, and didn't quite finish.  Now the refactor is complete, but I can see there is an argument to be made for not having a special `IsTime()` function, given that you could end up with a whole bunch of `IsSomeDataType()` functions moving forward.